frontend/src/components/smart/PvPArena.vue

```diff
@@ -212,8 +228,41 @@ export default {
       };
     },
 
-    kickCharacterFromArena() {
+    async kickCharacterFromArena() {
       this.isCharacterInArena = false;
+
+      const kickedEvents = await this.getKickedEvents(this.contracts());
+
+      if (kickedEvents.length) {
+        const formattedResult = formatCharacterKickedEvent(kickedEvents[kickedEvents.length - 1].returnValues);
+
+        if (+localStorage.getItem(`${getCharacterNameFromSeed(this.currentCharacterId)}: lastKickedTime`) < +formattedResult.timestamp) {
+          localStorage.setItem(`${getCharacterNameFromSeed(this.currentCharacterId)}: lastKickedTime`, formattedResult.timestamp);
+
+          this.recentlyKicked.characterId = formattedResult.characterId;
+          this.recentlyKicked.kickedBy = formattedResult.kickedBy;
+        }
+      }
+    },
+
+    async updateRank() {
+      this.characterInformation.rank = await this.contracts().PvpArena.methods.rankingPointsByCharacter(this.currentCharacterId)
+        .call({ from: this.defaultAccount });
+    },
+
+    handleCloseModal() {
+      this.recentlyKicked.characterId = null;
+      this.recentlyKicked.kickedBy = null;
+    },
+
+    async getKickedEvents(contracts) {
+      const kickedEvents = await contracts.PvpArena.getPastEvents('CharacterKicked', {
+        filter: { characterID: this.currentCharacterId },
+        toBlock: 'latest',
+        fromBlock: 'earliest'
```

Comment:
Can't  we scan from the previously stored timestamp? This will scan through the whole blockchain history lol